src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -1666,17 +1666,17 @@ type IncrementalBuilder(ctokCtor: CompilationThreadToken, frameworkTcImportsCach
 
     member this.IncrementUsageCount() = 
         assertNotDisposed() 
-        referenceCount  <- referenceCount  + 1
+        System.Threading.Interlocked.Increment(referenceCount) |> ignore
         { new System.IDisposable with member x.Dispose() = this.DecrementUsageCount() }
 
     member this.DecrementUsageCount() = 
         assertNotDisposed()
-        referenceCount  <- referenceCount  - 1
-        if referenceCount = 0 then 
+        System.Threading.Interlocked.Decrement(referenceCount) |> ignore
+        if !referenceCount = 0 then 
```

Comment:
This may race if someone else modifies it between these calls. I think the return value of ``Interlocked.Decrement`` should be used instead